### PR TITLE
Creating new Alert Troubleshooting section (JSC#667)

### PR DESCRIPTION
**Pull request by asettle**
A customer has requested that we provide further detail on the alerts and how to manage from a customer perspective. This is an exact copy of the upstream documentation as we are unable to link.

### CODE REVIEW

**Comment by asettle** [comment]
@tbazant @lproven @ajaeger  - this is line for line exactly what is in https://docs.ceph.com/docs/nautilus/rados/operations/health-checks/ as per the request from https://jira.suse.com/browse/SES-667

If you see anything that should really be edited, we should do it, but otherwise, a cursory glance so we can get this through the door (it's marked _Critical_) would be helpful. Thanks!

File: xml/admin_monitoring_alerting.xml [hunk 1]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><filename>POOL_TOO_FEW_PGS</filename></term>
+      <listitem>
+        <para>One or more pools should probably have more PGs, based on the
+          amount of data that is currently stored in the pool. This can lead
+          to suboptimal distribution and balance of data across the OSDs in
+          the cluster, and similarly reduce overall performance. This warning
+          is generated if the pg_autoscale_mode property on the pool is set to warn.
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
          is generated if the <option>pg_autoscale_mode</option> property on the pool is set to warn.
```

File: xml/admin_monitoring_alerting.xml [hunk 2]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+        <para>You can also manually set the number of PGs for the pool to the recommended amount with:</para>
+<screen>ceph osd pool set <replaceable>pool-name</replaceable> pg_num <replaceable>new-pg-num</replaceable></screen>
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><filename>TOO_MANY_PGS</filename></term>
+      <listitem>
+        <para>The number of PGs in use in the cluster is above the configurable
+          threshold of <literal>mon_max_pg_per_osd</literal> PGs per OSD. If this threshold is
+          exceed the cluster will not allow new pools to be created, pool <literal>pg_num</literal>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
          exceeded the cluster will not allow new pools to be created, pool <literal>pg_num</literal>
```

File: xml/admin_monitoring_alerting.xml [hunk 3]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+          to be increased, or pool replication to be increased (any of which
+          would lead to more PGs in the cluster). A large number of PGs can
+          lead to higher memory utilization for OSD daemons, slower peering
+          after cluster state changes (like OSD restarts, additions, or
+          removals), and higher load on the Manager and Monitor daemons.
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
          removals), and higher load on the &mgr; and &mon; daemons.
```

File: xml/admin_monitoring_alerting.xml [hunk 4]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+          The simplest way to mitigate the problem is to increase the number
+          of OSDs in the cluster by adding more hardware. Note that the OSD
+          count used for the purposes of this health check is the number
+          of “in” OSDs, so marking “out” OSDs “in” (if there are any) can also help:</para>
+<screen>ceph osd in <replaceable>osd id(s)</replaceable></screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
<screen>ceph osd in <replaceable>OSD_IDs</replaceable></screen>
```

File: xml/admin_monitoring_alerting.xml [hunk 5]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+      <listitem>
+        <para>One or more pools has an average number of objects per PG that
+          is significantly higher than the overall cluster average. The
+          specific threshold is controlled by the <literal>mon_pg_warn_max_object_skew</literal>
+          configuration value.
+          This is usually an indication that the pool(s) containing most of
+          the data in the cluster have too few PGs, and/or that other pools
+          that do not contain as much data have too many PGs.
+          The threshold can be raised to silence the health warning by
+          adjusting the <literal>mon_pg_warn_max_object_skew</literal> config option on the monitors.
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
          adjusting the <literal>mon_pg_warn_max_object_skew</literal> configuration option on the monitors.
```

File: xml/admin_monitoring_alerting.xml [hunk 6]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+    <varlistentry>
+      <term><filename>POOL_APP_NOT_ENABLED</filename></term>
+      <listitem>
+        <para>A pool exists that contains one or more objects but has not
+          been tagged for use by a particular application.
+          Resolve this warning by labeling the pool for use by an application.
+          For example, if the pool is used by RBD:
+        </para>
+<screen>rbd pool init <replaceable>poolname</replaceable></screen>
+        <para>If the pool is being used by a custom application ‘foo’,
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
        <para>If the pool is being used by a custom application <replaceable>FOO</replaceable>,
```

File: xml/admin_monitoring_alerting.xml [hunk 7]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+          you can also label via the low-level command:</para>
+<screen>ceph osd pool application enable foo</screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
<screen>ceph osd pool application enable <replaceable>FOO</replaceable></screen>
```

File: xml/admin_monitoring_alerting.xml [hunk 8]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+        <para>One or more objects in the cluster cannot be found. Specifically,
+          the OSDs know that a new or updated copy of an object should exist,
+          but a copy of that version of the object has not been found on OSDs
+          that are currently online.
+          Read or write requests to unfound objects will block.
+          Ideally, a down OSD can be brought back online that has the more
+          recent copy of the unfound object. Candidate OSDs can be identified
+          from the peering state for the PG(s) responsible for the unfound object:
+        </para>
+<screen>ceph tell <replaceable>pgid</replaceable> query</screen>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
<screen>ceph tell <replaceable>PG_ID</replaceable> query</screen>
```

File: xml/admin_monitoring_alerting.xml [hunk 9]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+        </para>
+<screen>ceph pg deep-scrub <replaceable>pgid</replaceable></screen>
+      </listitem>
+    </varlistentry>
+  </variablelist>
+  <variablelist><title>MISCELLANEOUS</title>
+  <varlistentry>
+    <term><filename>RECENT_CRASH</filename></term>
+    <listitem>
+      <para>One or more Ceph daemons has crashed recently, and the
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
      <para>One or more &ceph; daemons has crashed recently, and the
```

File: xml/admin_monitoring_alerting.xml [hunk 10]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+        “recent” means is controlled by the option
+        <literal>mgr/crash/warn_recent_interval</literal> (default: two weeks).
+        These warnings can be disabled entirely with:</para>
+<screen>ceph config set mgr/crash/warn_recent_interval 0</screen>
+    </listitem>
+  </varlistentry>
+    <varlistentry>
+      <term><filename>TELEMETRY_CHANGED</filename></term>
+      <listitem>
+        <para>Telemetry has been enabled, but the contents of the telemetry
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]: maybe add an xref to a telemtry module section?

File: xml/admin_monitoring_alerting.xml [hunk 11]
```diff
@@ -1062,5 +1062,1045 @@ monitoring:
...
+          report have changed since that time, so telemetry reports will not be sent.
+          The Ceph developers periodically revise the telemetry feature to
+          include new and useful information, or to remove information found
+          to be useless or sensitive. If any new information is included in
+          the report, Ceph will require the administrator to re-enable
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
          the report, &ceph; will require the administrator to re-enable
```

**Pull request merged**